README.md

```diff
@@ -1,2 +1,88 @@
-# hexabot-plugin-chatgpt
-The OpenAI ChatGPT Plugin for Hexabot Chatbot / Agent Builder to enable the LLM RAG Capability
+# ChatGPT RAG Plugin for Hexabot Chatbot Builder
+
+The ChatGPT RAG Plugin integrates OpenAI’s GPT-4 model with Hexabot chatbot workflows to enable Retrieval-Augmented Generation (RAG) capabilities. This plugin allows users to enrich their chatbot responses using relevant content retrieved from a content management system, making interactions more contextual and accurate.
+
+[Hexabot](https://hexabot.ai/) is an open-source chatbot / agent solution that allows users to create and manage AI-powered, multi-channel, and multilingual chatbots with ease. If you would like to learn more, please visit the [official github repo](https://github.com/Hexastack/Hexabot/).
+
+## Prerequisites
+
+Before setting up the ChatGPT RAG Plugin, you will need to generate an API token from OpenAI.
+
+1. Go to the [OpenAI API page](https://openai.com/api).
+2. Obtain your API key.
+3. Once you have your API key, you can configure the plugin in Hexabot.
+
+## Installation
+
+First, navigate to your Hexabot project directory and make sure the dependencies are installed:
+
+```sh
+cd ~/projects/Hexabot
+npm install
+```
+
+To install the ChatGPT RAG Plugin, run the following command:
+
+```sh
+npx hexabot install plugin Hexastack/hexabot-plugin-chatgpt-rag
+```
+
+## Configuration
+
+The ChatGPT RAG Plugin provides several customizable settings that can be configured through the Hexabot admin interface:
+
+- **Token**: Your OpenAI API token. This is required for authentication.
+- **Model**: The model to be used for generating responses (e.g., `gpt-4o-mini`).
+- **Response Size**: The maximum number of tokens in the AI response.
+- **Messages to Retrieve**: The number of recent messages to include as context when making requests to OpenAI.
+- **Context**: A description or context of the chatbot’s role to guide its responses.
+- **Instructions**: Custom instructions for how the AI should handle responses based on your use case.
+
+## How to Use
+
+1. Access the Hexabot Visual Editor.
+2. Drag the ChatGPT RAG block from "Custom Blocks" onto the canvas.
+3. Double-click the block to edit and configure the plugin’s settings, including the API token, model, and context.
+
+## Example
+
+Here’s an example prompt generated by the plugin when sending a request to OpenAI:
+
+```
+CONTEXT: You are an AI Chatbot that works for Hexastack. Here is a description of the company: ...
+
+DOCUMENTS:
+DOCUMENT 0
+Title: Example Title
+Data: Example Data...
+RECENT MESSAGES:
+
+- Message 1: Hi, how can I assist you today?
+- Message 2: Can you explain more about Hexastack’s services?
```

Comment:
Is this the way OpenAI ChatGPT recommends in their documentation ?

Reply:
This example inspired from the [documentation](https://platform.openai.com/docs/examples).